File: src/Cocoar.SignalARRR.Swift/Sources/CocoarSignalARRR/MethodNames.swift
import Foundation

/// String constants matching `Cocoar.SignalARRR.Common.Constants.MethodNames`.
///
/// These are the SignalR hub method names used by the SignalARRR wire protocol.
public enum MethodNames {
    // MARK: - Client → Server

    public static let invokeMessageOnServer = "InvokeMessage"
    public static let invokeMessageResultOnServer = "InvokeMessageResult"
    public static let sendMessageToServer = "SendMessage"
    public static let streamMessageFromServer = "StreamMessage"

    // MARK: - Server → Client

    public static let invokeServerRequest = "InvokeServerRequest"
    public static let replyServerRequest = "ReplyServerRequest"
    public static let challengeAuthentication = "ChallengeAuthentication"
    public static let invokeServerMessage = "InvokeServerMessage"
    public static let cancelTokenFromServer = "CancelTokenFromServer"

    // MARK: - Streaming

    public static let streamItemToServer = "StreamItemToServer"
    public static let streamCompleteToServer = "StreamCompleteToServer"
}

File: src/Cocoar.SignalARRR.Common/Constants/MethodNames.cs
namespace Cocoar.SignalARRR.Common.Constants {

    /// <summary>
    /// The SignalR method names SignalARRR sends and listens on. They are the wire protocol, so they
    /// are constants — two of them used to have public setters, which let any consumer rewrite the
    /// protocol process-wide at runtime and leave every other client on the connection talking a
    /// name nothing was listening for.
    /// </summary>
    public static class MethodNames {
        public const string InvokeMessageOnServer = "InvokeMessage";
        public const string InvokeMessageResultOnServer = "InvokeMessageResult";
        public const string SendMessageToServer = "SendMessage";
        public const string StreamMessageFromServer = "StreamMessage";

        public const string InvokeServerRequest = "InvokeServerRequest";

        public const string ChallengeAuthentication = "ChallengeAuthentication";
        public const string InvokeServerMessage = "InvokeServerMessage";

        public const string CancelTokenFromServer = "CancelTokenFromServer";

        public const string StreamItemToServer = "StreamItemToServer";
        public const string StreamCompleteToServer = "StreamCompleteToServer";
    }
}
